templates/daemon.systemd.erb

```diff
@@ -14,6 +14,9 @@ EnvironmentFile=<%= @env_file_path %>/<%= @name%>
 ExecStart=<%= @bin_dir %>/<%= @bin_name %><% for option in Shellwords.split(@options) %> \
 <%= option -%>
 <% end %>
+<%- @systemd_configs.each_pair do |key,value| -%>
```

Comment:
instead of extending the erb template, what do you think about using https://github.com/camptocamp/puppet-systemd#drop-in-files ? there are probably many people where the installation method is package, and we don't manage the systemd file. but the people still want to overwrite it. a drop-in file would be a generic solution for that?

Reply:
Ha, it's funny that did not come to my mind, I make heavy use of drop in files pretty much everywhere else.  For some reason I immediately went to needing to modify this template.

Reply:
Would it make sense to add the dropin here or just dump it into a local profile?

Reply:
mhm good question! maybe a profile isn't a bad idea. We could add a new parameter to prometheus::daemon with an array of strings. in a dropin file we could iterate on it, but we would also need to add the new parameter to all exporter classes in the modules (with puppet-strings docs). If you want to do the work I'm happy to review it, but keeping the dropin in your profile might be way easier to implement.

Reply:
The exporter I needed to do this on I ended up finding a different solution that didn't require a dropin, so at this time I have no need for this change.  I'll go ahead and close this out.